GVFS/GVFS/RepairJobs/GitHeadRepairJob.cs

```diff
@@ -31,114 +30,36 @@ public override IssueType HasIssue(List<string> messages)
-        {
-            string refLogPath = Path.Combine(enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Logs.Root, fullSymbolicRef);
-            if (!File.Exists(refLogPath))
-            {
-                refLog = null;
-                error = "Could not find reflog for ref '" + fullSymbolicRef + "'";
-                return false;
-            }
+                this.Tracer.RelatedEvent(
+                    EventLevel.Informational,
+                    "MovedHead",
+                    new EventMetadata
+                    {
+                        { "DestinationCommit", newHeadSha }
+                    });
 
-            try
-            {
-                string refLogContents = File.ReadLines(refLogPath).Last();
-                if (!RefLogEntry.TryParse(refLogContents, out refLog))
-                {
-                    error = "Last ref log entry for " + fullSymbolicRef + " is unparsable.";
-                    return false;
-                }
-            }
-            catch (IOException ex)
-            {
-                refLog = null;
-                error = "IOException while reading reflog '" + refLogPath + "': " + ex.Message;
-                return false;
+                messages.Add("As a result of the repair, 'git status' will now complain that HEAD is detached");
+                messages.Add("You can fix this by creating a branch using 'git checkout -b <branchName>'");
             }
 
-            error = null;
-            return true;
+            return result;
         }
-        
-        private static bool TryParseHead(Enlistment enlistment, List<string> messages)
-        {
-            string refPath = Path.Combine(enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Head);
-            if (!File.Exists(refPath))
-            {
-                messages.Add("Could not find ref file for '" + GVFSConstants.DotGit.Head + "'");
-                return false;
-            }
-
-            string refContents;
-            try
-            {
-                refContents = File.ReadAllText(refPath).Trim();
-            }
-            catch (IOException ex)
-            {
-                messages.Add("IOException while reading .git\\HEAD: " + ex.Message);
-                return false;
-            }
 
-            const string MinimallyValidRef = "ref: refs/";
-            if (refContents.StartsWith(MinimallyValidRef, StringComparison.OrdinalIgnoreCase) ||
-                SHA1Util.IsValidShaFormat(refContents))
-            {
-                return true;
-            }
-
-            messages.Add("Invalid contents found in '" + GVFSConstants.DotGit.Head + "': " + refContents);
-            return false;
+        protected override IEnumerable<string> GetRefs()
+        {
+            return new[] { GVFSConstants.DotGit.HeadName };
         }
 
         private bool CanBeRepaired(List<string> messages)

```

Comment:
Why didn't `CanBeRepaired` make it into the base class?

Reply:
What if HEAD is pointing to the ref that you're currently fixing? In effect, you'd be changing HEAD during one of those conflict states.